File: ai/src/main/scala/org/pingel/gestalt/ui/Widget.java
package org.pingel.gestalt.ui;

import java.awt.Point;
import java.awt.event.MouseEvent;

import org.pingel.gestalt.core.History;
import org.pingel.gestalt.core.Lexicon;

public interface Widget {
    
    public Widget mousePressed(MouseEvent e, History history, Lexicon lookupLexicon, Lexicon newLexicon);

    public boolean mouseClicked(MouseEvent e, History history, Lexicon lexicon);
    
    public void release(Point p, History history, Lexicon lookupLexicon, Lexicon newlexicon);
    
    public void drag(Point p, History history, Lexicon lexicon);
    
    public Point getCenter();
    
    public void setHighlighted(boolean h);
 
    public Point getBounds();

}

File: gestalt/src/main/scala/org/pingel/gestalt/ui/Widget.scala
package org.pingel.gestalt.ui

import java.awt.Point
import java.awt.event.MouseEvent

import org.pingel.gestalt.core.History
import org.pingel.gestalt.core.Lexicon

trait Widget {
    
	def mousePressed(e: MouseEvent, history: History, lookupLexicon: Lexicon, newLexicon: Lexicon): Widget

	def mouseClicked(e: MouseEvent, history: History, lexicon: Lexicon): Boolean
    
	def release(p: Point, history: History, lookupLexicon: Lexicon, newlexicon: Lexicon): Unit
    
	def drag(p: Point, history: History, lexicon: Lexicon): Unit
    
	def getCenter(): Point
    
	def setHighlighted(h: Boolean): Unit
 
	def getBounds(): Point

}
